src/tallies/TallyBase.C

```diff
@@ -329,7 +344,7 @@ TallyBase::applyTriggersToLocalTally(openmc::Tally * tally)
     for (int score = 0; score < _tally_score.size(); ++score)
       tally->triggers_.push_back({_openmc_problem.triggerMetric((*_tally_trigger)[score]),
                                   _tally_trigger_threshold[score],
```

Comment:
I may have missed this in an earlier PR, but what happens if the user does not set the `trigger_threshold`? In the constructor, a value is only put into it if the user provided it, but here it's used regardless. Is there a loophole here which would result in a segfault/something else going wrong?

Reply:
I don't believe it's possible for this to segfault as `_tally_trigger_threshold` is default constructed with a length of zero. If the user doesn't set `trigger_threshold` then the length of `_tally_trigger_threshold` will not equal the length of `_tally_score`, so an error will get thrown here:
```
if (_tally_trigger_threshold.size() != _tally_score.size())
      mooseError("'trigger_threshold' (size " + std::to_string(_tally_trigger_threshold.size()) +
                 ") must have the same length as 'score' (size " +
                 std::to_string(_tally_score.size()) + ")");
```

The error is a little unintuitive since the user might not know that they need to also provide a value for  `trigger_threshold`, so I've added a `paramError` which should make it easier for the user to figure out what they need to set to use triggers.